azurerm/internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -661,24 +679,25 @@ func resourceArmCosmosDbAccountDelete(d *schema.ResourceData, meta interface{})
 		Refresh: func() (interface{}, string, error) {
 			resp, err2 := client.Get(ctx, resourceGroup, name)
 			if err2 != nil {
-				if utils.ResponseWasNotFound(resp.Response) {
-					return resp, "NotFound", nil
-				}
 				return nil, "", fmt.Errorf("Error reading CosmosDB Account %q after delete (Resource Group %q): %+v", name, resourceGroup, err2)
 			}
 
+			if utils.ResponseWasNotFound(resp.Response) {
+				return resp, "NotFound", nil
+			}
+
```

Comment:
TODO When #7616 is merged, this block will need to be relocated as 404s will raise errors again on Database Account GET https://github.com/Azure/azure-sdk-for-go/blame/v44.0.0/services/cosmos-db/mgmt/2020-04-01/documentdb/databaseaccounts.go#L499